modules/frontend/search_sharder.go

```diff
@@ -225,13 +225,42 @@ func (s *asyncSearchSharder) ingesterRequest(ctx context.Context, tenantID strin
 
 	// if ingester start == ingester end then we don't need to query it
 	if ingesterStart == ingesterEnd {
-		return nil, nil
+		return nil
 	}
 
 	searchReq.Start = ingesterStart
 	searchReq.End = ingesterEnd
 
-	return buildIngesterRequest(ctx, tenantID, parent, &searchReq)
+	// Split the start and end range into sub requests for each range.
+	shards := s.cfg.IngesterShards
+	duration := searchReq.End - searchReq.Start
+	interval := duration / uint32(s.cfg.IngesterShards)
+	// If the interval is less than a minute, dont' shard
+	if interval < 60 {
```

Comment:
I'm not sure if this is right.  Perhaps we force-reduce the shard count until the interval is above a minute.  Using 1 minute here as a notion of "doing too much work" on overlapping blocks.  Should this be configurable?